pkg/kudoctl/cmd/uninstall.go

```diff
@@ -0,0 +1,128 @@
+package cmd
+
+import (
+	"fmt"
+
+	"github.com/kudobuilder/kudo/pkg/kudoctl/clog"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/env"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/util/kudo"
+
+	"github.com/spf13/cobra"
+)
+
+const (
+	uninstallExample = `kubectl kudo uninstall --instance flink`
+)
+
+type uninstallOptions struct {
+	InstanceName string
+	Purge        bool
+}
+
+type uninstallCmd struct{}
+
+func (cmd *uninstallCmd) run(options uninstallOptions, settings *env.Settings) error {
+	kc, err := kudo.NewClient(settings.Namespace, settings.KubeConfig)
+	clog.V(3).Printf("acquiring kudo client")
+	if err != nil {
+		clog.V(3).Printf("failed to acquire kudo client")
+		return fmt.Errorf("failed to acquire kudo client: %w", err)
+	}
+
+	return cmd.uninstall(kc, options.InstanceName, options.Purge, settings)
+}
+
+func (cmd *uninstallCmd) uninstall(kc *kudo.Client, instanceName string, purge bool, settings *env.Settings) error {
+	instance, err := kc.GetInstance(instanceName, settings.Namespace)
+	if err != nil {
+		return fmt.Errorf("failed to verify if instance already exists: %w", err)
+	}
+
+	if instance == nil {
+		return fmt.Errorf("instance %s in namespace %s does not exist in the cluster", instanceName, settings.Namespace)
```

Comment:
Returning an error here means that this operation isn't idempotent. I'm following the example of `kudo install` here which also isn't idempotent.

Reply:
maybe we make this a warning and make it idempotent.

Reply:
How about having a follow-up PR that changes this here as well as for `kudo install`? There might be other commands as well that aren't idempotent yet and IMO we should make this a consistent change while we can still break things.

Reply:
Created https://github.com/kudobuilder/kudo/issues/952 for this.

Reply:
Update: The operation is indeed idempotent. It's more of a question on how this is surfaced as part of the CLI. Given that the [Kubernetes API would surface this as an error](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#idempotency), we should follow that example here as well. Which we do with returning an error here.